src/main.js

```diff
@@ -104,120 +123,138 @@ export default class Auth {
 	 */
-	api(endpoint, body) {
+	async api(endpoint, body) {
 		const url =
 			endpoint === 'token'
 				? `https://securetoken.googleapis.com/v1/token?key=${this.apiKey}`
 				: `https://identitytoolkit.googleapis.com/v1/accounts:${endpoint}?key=${this.apiKey}`;
 
-		return fetch(url, {
+		const response = await fetch(url, {
 			method: 'POST',
-			body: typeof body === 'string' ? body : JSON.stringify(body)
-		}).then(async response => {
-			let data = await response.json();
-
-			// If the response returned an error, try to get a Firebase error code/message.
-			// Sometimes the error codes are joined with an explanation, we don't need that(its a bug).
-			// So we remove the unnecessary part.
-			if (!response.ok) {
-				const code = data.error.message.replace(/: [\w ,.'"()]+$/, '');
-				throw Error(code);
-			}
+			body: typeof body === 'string' ? body : JSON.stringify(body),
+		});
+		let data = await response.json();
+		// If the response returned an error, try to get a Firebase error code/message.
+		// Sometimes the error codes are joined with an explanation, and we don't need that (it's a bug).
+		// So we remove the unnecessary part.
+		if (!response.ok) {
+			const code = data.error.message.replace(/: [\w ,.'"()]+$/, '');
+			throw Error(code);
+		}
 
-			// Add a hidden date property to the returned object.
-			// Used mostly to calculate the expiration date for tokens.
-			Object.defineProperty(data, 'expiresAt', { value: Date.parse(response.headers.get('date')) + 3600 * 1000 });
-			return data;
+		// Calculate the expiration date for tokens.
+		Object.defineProperty(data, 'expiresAt', {
+			value: Date.parse(response.headers.get('date')) + 3600 * 1000,
 		});
+
+		return data;
 	}
 
 	/**
-	 * Makes sure the user is logged in and has up-to-date credentials.
-	 * @throws Will throw if the user is not logged in.
+	 * Makes sure the user is signed-in and has up-to-date credentials.
+	 * @throws Will throw if the user is not signed-in.
 	 * @private
 	 */
 	async enforceAuth() {
-		if (!this.user) throw Error('The user must be logged-in to use this method.');
-		return this.refreshIdToken(); // Won't do anything if the token is valid.
+		if (this.user) {
+			return this.refreshIdToken(); // Won't do anything if the token is valid.
+		} else {
+			throw Error('The user must be signed-in to use this method.');
+		}
 	}
 
 	/**
-	 * Updates the user data in the localStorage.
-	 * @param {Object} userData the new user data.
-	 * @param {boolean} [updateStorage = true] Whether to update local storage or not.
+	 * Updates the user data in localStorage.
+	 * @param {Object} userData New user data.
+	 * @param {boolean} [updateStorage = true] Check whether to update localStorage or not.
 	 * @private
 	 */
 	async setState(userData, persist = true, emit = true) {
 		this.user = userData;
-		persist && (await this.storage[userData ? 'set' : 'remove'](this.sKey('User'), JSON.stringify(userData)));
-		emit && this.emit();
+		if (persist) {
```

Comment:
`&&` takes fewer characters than `if(x) {}`.
It is less idiomatic, I get it, but bear in mind that this code is aggressively optimized for space.